docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -313,121 +285,100 @@ class ComponentExample extends React.Component<ComponentExampleProps, ComponentE
 
   getDisplayName = () => this.props.examplePath.split('/')[1]
 
-  handleChangeCode = (sourceCode: string) => {
-    this.sourceCodeMgr.currentCode = sourceCode
-    this.updateAndRenderSourceCode()
+  handleCodeApiChange = apiType => () => {
+    this.props.handleCodeAPIChange(apiType)
   }
 
-  updateAndRenderSourceCode = () => {
-    this.setState({ sourceCode: this.sourceCodeMgr.currentCode })
-  }
+  handleCodeLanguageChange = language => () => {
+    const { handleCodeLanguageChange, wasCodeChanged } = this.props
 
-  setApiCodeType = (codeType: SourceCodeType) => {
-    this.sourceCodeMgr.codeType = codeType
-    this.updateAndRenderSourceCode()
+    if (wasCodeChanged) {
+      if (confirm('Lose your changes?')) {
```

Comment:
It was awkward to me to lose the changes between switching to Javascript and Typescript. Can't we store both source codes in state? Not sure if this will really be used by the users, but just wondering...

Reply:
I want to avoid implementation complications there.

If we will get claims about it, we can refactor `ComponentSourceManager` somehow to store both source codes.

Reply:
Valid point..